docs/guides/using-watt-with-node-config.md

```diff
@@ -0,0 +1,178 @@
+# Using Watt With Node Config
+
+[Node-config](https://www.npmjs.com/package/config) is a popular configuration management package that helps organize settings across different deployment environments in your application. It creates a unified configuration system that works seamlessly with both [Watt](https://platformatic.dev/docs/watt/overview) and other `npm` modules.
+When building a Watt application with multiple services, each service can maintain its own independent configuration using `node-config`. This allows different services to use different environment configurations as needed.
+
+
+
+## Installation and Setup
+
+First, install `node-config` in the root of your Watt application:
+
+```sh
+npm install config
+```
+
+Create a `config` directory in your `service` folder and set up your default configuration for each service:
+
+```sh
+mkdir config
+touch config/default.json
+```
+
+In `{service}/config/default.json`, add your base configuration:
+
+```sh
+{
+  "foo": "bar"
+}
+```
+
+## Environment-specific configuration 
+
+For development-specific settings, create a separate configuration file:
+
+```sh
+touch config/dev.json
+```
+
+In `{service}/config/dev.json`, override any default values:
+
+```sh
+{
+  "foo": "baz"
+}
+```
+
+## Configuration structure
+
+Instead of using simple key-value pairs, consider organizing your configurations more systematically, below is an example:
+
+```sh
+{
+  "server": {
+    "port": 3000,
+    "host": "localhost"
+  },
+  "api": {
+    "endpoint": "http://api.example.com",
+    "timeout": 5000
+  },
+  "database": {
+    "url": "mongodb://localhost:27017",
+    "name": "myapp"
+  }
+}
+```
+
+:::important
+It's important to note that for a secure configuration, use your environment variables for your application  secrets and validate your configuration values when you run your application. 
+:::
+
+## Service-Specific Configuration 
+
+You can configure each [service](https://platformatic.dev/docs/service/overview) environment variables in your Watt configuration file:
+
+```json
+{
+  "services": [
+    {
+      "id": "service-a",
+      "path": "./services/service-a",
+      "env": {
+        "NODE_CONFIG_DIR": "./services/service-a/config",
+        "NODE_ENV": "development"
+      }
+    },
+    {
+      "id": "service-b",
+      "path": "./services/service-b",
+      "env": {
+        "NODE_CONFIG_DIR": "./services/service-b/config",
+        "NODE_ENV": "production"
```

Comment:
as an additional step, show how to
use platformatic .env interpolation to remap a SERVICE_A_NODE_ENV to NODE_ENV